src/lib.rs

```diff
@@ -359,6 +365,15 @@ pub mod auth {
         }
     }
 
+    #[pg_extern(parallel_safe, stable)]
+    pub fn uid() -> Option<pgrx::Uuid> {
+        let user_id = user_id()?;
+        if let Ok(uuid) = Uuid::parse_str(&user_id) {
+            return Some(PgUuid::from_bytes(*uuid.as_bytes()));
+        }
+        None
+    }
+
```

Comment:
nit: this can be a bit more idiomatic

```suggestion
    #[pg_extern(parallel_safe, stable)]
    pub fn uid() -> Option<pgrx::Uuid> {
        user_id()
            .and_then(|uuid| Uuid::parse_str(&uuid).ok())
            .map(|uuid| PgUuid::from_bytes(*uuid.as_bytes()))
    }
```